apps/condo/domains/meter/schema/CommunalMeterReading.test.js

```diff
@@ -0,0 +1,236 @@
+                const client = await makeClientWithNewRegisteredAndLoggedInUser()
+
+                const [organization] = await createTestOrganization(admin)
+                const [property] = await createTestProperty(admin, organization, { map: buildingMapJson })
+                const [role] = await createTestOrganizationEmployeeRole(admin, organization, {
+                    canManageMeterReadings: true,
+                })
+
+                await createTestOrganizationEmployee(admin, organization, client.user, role)
+                const [meter] = await createTestCommunalMeter(client, organization, property, resource, {})
+
+                const [meterReading] = await createTestCommunalMeterReading(client, meter, source)
+
+                expect(meterReading.id).toMatch(UUID_RE)
+            })
+
+            test('employee without "canManageMeters" role: cannot create MeterReadings', async () => {
+                const client = await makeClientWithNewRegisteredAndLoggedInUser()
+
+                const [organization] = await createTestOrganization(admin)
+                const [property] = await createTestProperty(admin, organization, { map: buildingMapJson })
+                const [role] = await createTestOrganizationEmployeeRole(admin, organization, {
+                    canManageMeterReadings: false,
+                })
+
+                await createTestOrganizationEmployee(admin, organization, client.user, role)
+                const [meter] = await createTestCommunalMeter(client, organization, property, resource, {})
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await createTestCommunalMeterReading(client, meter, source)
+                })
+            })
+
+            test('anonymous can\'t', async () => {
+                const client = await makeClient()
+
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await createTestCommunalMeterReading(client, { id: 'id' }, { id: 'id' })
+                })
+            })
+        })
+
+        describe('update', () => {
+            test('admin can', async () => {
+                const [organization] = await createTestOrganization(admin)
+                const [property] = await createTestProperty(admin, organization, { map: buildingMapJson })
+                const [meter] = await createTestCommunalMeter(admin, organization, property, resource, {})
+                const [meterReading] = await createTestCommunalMeterReading(admin, meter, source)
+
+                const [obj, attrs] = await updateTestCommunalMeterReading(admin, meterReading.id)
+
+                expect(obj.dv).toEqual(1)
+                expect(obj.sender).toEqual(attrs.sender)
+                expect(obj.v).toEqual(2)
+                expect(obj.updatedBy).toEqual(expect.objectContaining({ id: admin.user.id }))
+            })
+
+            test('employee with canManageMeterReadings cannot', async () => {
```

Comment:
You can also move employee with / without rights, support and anon to actors initialisation stage, so you can speed up your tests significantly

Reply:
Then another question. Do we need tests that test everything on the perfect data we just created? And this way the tests become more comprehensive, because in addition to the data from the tests, the organization can have other data and interactions, which brings them closer to real life